### DISCUSSION TITLE
how to build signed air package mobile project for installation on desktop?

### DISCUSSION THREAD

**Discussion by jebarbieri** [BowlerHatLLC/vscode-as3mxml#662] [General]
how to build signed air package mobile project for installation on desktop? In flashbuilder has this option. it's possible?

![build](https://user-images.githubusercontent.com/38330359/209852476-e4dc3f90-9672-4de5-86e4-c8571e5a032d.png)

**Comment by joshtynjala** [comment]
You need to specify your app's [`signingOptions`](https://github.com/BowlerHatLLC/vscode-as3mxml/wiki/asconfig.json#signing-options) in your _asconfig.json_ file. Example:

```jsonc
"airOptions": {
	"output": "bin/MyApp.air",
	"signingOptions": {
		"storetype": "pkcs12",
		"keystore": "path/to/certificate.p12"
	}
}
```

Then, run the appropriate [AIR desktop packaging task](https://github.com/BowlerHatLLC/vscode-as3mxml/wiki/Package-an-Adobe-AIR-application-in-Visual-Studio-Code#air-desktop). It will ask for your certificate's password during the build process.

**Reply by jebarbieri** [reply]
### **I created a air project and set signingOptions for windows:**
 
```jsonc
"config": "air"
```

```jsonc
"windows": {
			"output": "bin/atendi-mobile.air",
			"signingOptions": {
				"storetype": "pkcs12",
				"keystore": "cert/desktop/atendi_desktop.p12"
			}
		}, 
```

The project  with these settings, does not find the mobile classes:
- spark.skins.mobile:TextInputSkin
- spark.skins.mobile.supportClasses.ButtonSkinBase

### **How to add them to the project?**

**Reply by joshtynjala** [reply]
When you create a mobile app, you would normally use `"config": "airmobile"` instead. However, it looks like are creating a desktop app, but using the mobile framework/theme, for some reason. I believe that there are one or more compiler options that you must use to allow a Flex desktop app to use the mobile theme, but I do not recall exactly what is necessary. I've never actually done that in any of my own projects. I've only seen that others have figured out how to do it.

**Reply by joshtynjala** [reply]
If you still have access to Flash Builder, you could try adding the `dump-config` compiler option to have Flash Builder write all of the compiler options it uses to an XML file. Then, you might be able to figure out which ones are required.

```
--dump-config+=path/to/file.xml
```

**Comment by jebarbieri** [comment]
Problem solved. I added this tag in air.config
       
 
       <theme>
         <filename>themes/Mobile/mobile.swc</filename>
      </theme>
